walkthroughs/ecs-service-connectivity/yelb/deployments/platformdeployment/AWS/ECS/yelb-cloudformation-ECS-AppMesh-deployment.yaml

```diff
@@ -0,0 +1,631 @@
+               - EC2
+               - FARGATE
+
+    PublicIP:
+        Description: Please provide IP connectivity option
+        Type: String
+        Default: ENABLED
+        AllowedValues:
+               - ENABLED
+               - DISABLED
+
+
+Resources:
+                        
+    ServiceYelbAppserver:
+        Type: AWS::ECS::Service
+        Properties:
+            LaunchType: !Ref LaunchType
+            Cluster: !Ref Cluster
+            DesiredCount: !Ref CountOfAppserverTasks
+            ServiceRegistries:
+                - RegistryArn: !GetAtt YelbAppserverServiceDiscoveryEntry.Arn
+            TaskDefinition: !Ref 'TaskDefinitionYelbAppserver'
+            NetworkConfiguration:
+                AwsvpcConfiguration:
+                    AssignPublicIp: !Ref PublicIP
+                    Subnets: [ !Ref 'PublicSubnetOne' , !Ref 'PublicSubnetTwo'  ]
+                    SecurityGroups: [!Ref 'YelbAppserverSecurityGroup' ]
+                        
+    TaskDefinitionYelbAppserver:
+        Type: AWS::ECS::TaskDefinition
+        Properties:
+            Family: yelb-appserver
+            NetworkMode: awsvpc
+            RequiresCompatibilities:
+                - FARGATE
+            TaskRoleArn: !Ref YelbECSTaskIamRole
+            ExecutionRoleArn: !Ref 'YelbECSTaskExecutionRole'
+            Cpu: 256
+            Memory: 512
+            ProxyConfiguration:
+                Type: APPMESH
+                ContainerName: envoy
+                ProxyConfigurationProperties:
+                    - Name: IgnoredUID
+                      Value: '1337'
+                    - Name: ProxyIngressPort
+                      Value: '15000'
+                    - Name: ProxyEgressPort
+                      Value: '15001'
+                    - Name: AppPorts
+                      Value: '4567'
+                    - Name: EgressIgnoredIPs
+                      Value: '169.254.170.2,169.254.169.254'
+            ContainerDefinitions:
+                - Name: yelb-appserver
+                  Essential: true
+                  Image: !Ref YelbAppServerImage
+                  Environment:
+                    - Name: SEARCH_DOMAIN
+                      Value: !Ref 'Domain'
+                    - Name: RECIPE_API_ENDPOINT
+                      Value: 'http://www.recipepuppy.com/api/?q='
+                    - Name: YELB_DB_SERVER_ENDPOINT
+                      Value: !GetAtt YelbDb.Endpoint.Address
+                    - Name: YELB_REDIS_SERVER_ENDPOINT
+                      Value: !GetAtt YelbRedis.RedisEndpoint.Address
+                  LogConfiguration:
+                    LogDriver: awslogs
+                    Options:
+                        awslogs-group: !Ref AWS::StackName
+                        awslogs-region: !Ref AWS::Region
+                        awslogs-stream-prefix: yelb-appserver
+                  DependsOn:
+                      - ContainerName: envoy
+                        Condition: HEALTHY
```

Comment:
This dependency should be in the other direction: The Envoy should wait for the application to be healthy before starting. This ensures that the proxy doesn't begin receiving traffic before the application it's proxying to is ready.

Reply:
Followed Kiran Meduri's template https://github.com/kiranmeduri/yelb/blob/master/deployments/platformdeployment/AWS/ECS/yelb-cloudformation-ECS-AppMesh-deployment.yaml

Reply:
That template is wrong too actually. The envoy should be started last in the task to avoid 5xx responses when the application is not available yet.